packages/api/src/libs/zrx.e2e.ts

```diff
@@ -0,0 +1,39 @@
+import assert from "assert";
+import Client from "./zrx";
+
+const baseUrl = "https://api.0x.org/";
+describe("0x client", function () {
+  let client: Client;
+  it("should init", function () {
+    client = new Client(baseUrl);
+    assert.ok(client);
+  });
+  it("should get a price", async function () {
+    const params = {
+      sellToken: "WETH",
+      // show we can use addresses as well
+      buyToken:'0x6b175474e89094c44da98b954eedeac495271d0f', // DAI
+      sellAmount: "1000000000000000000",
+    };
+    const result = await client.price(params);
+    console.log(result)
+    assert.ok(result.price);
+  });
+  it("should have a human readable error", async function () {
+    const params = {
+      sellToken: "WETH",
+      buyToken: "DAI",
+      // missing sellAmount or buyAmount
+    };
+    const errorMessage =
+      "Bad Request(400): Validation Failed: should have required property 'sellAmount', should have required property 'buyAmount', should match exactly one schema in oneOf";
+    let plan = 1;
```

Comment:
What does `plan` stand for?

Reply:
plan represents how many tests we plan to make. so i count down every time i expect an assert to hit. just to make sure nothing gets skipped over. plan should be 0 at the end of the test.

Reply:
numExpectedErrors might be clearer.

Also, using a .catch on the promise that returns the error object might be easier to test against than counting errors using a try catch.

```js
assert.equal(await foo.bar().catch(e => e.message), "some message");
```

Reply:
nice i like that